crypto/batchverifier.c

```diff
@@ -0,0 +1,28 @@
+#include "sodium.h"
+int ed25519_batch_wrapper(const unsigned char *messages1D,
+                          const unsigned long long *mlen,
+                          const unsigned char *publicKeys1D,
+                          const unsigned char *signatures1D,
+                          size_t num,
+                          int *valid) {
+    int ret;
+    const unsigned char **messages2D, **publicKeys2D, **signatures2D;
+    messages2D = malloc(num * sizeof(unsigned char *));
+    publicKeys2D = malloc(num * sizeof(unsigned char *));
+    signatures2D = malloc(num * sizeof(unsigned char *));
```

Comment:
I can _absolutely_ see why we might prefer not to do this, but we could malloc all this space in one chunk (3 times bigger), and then just set `publicKey2D` and `signatures2D` into it at the 1/3 and 2/3 point.

I normally would think that's a little gross. But it's basically the same idea as we're doing for all of the pieces, right?